File: src/main/java/com/github/direnvloader/DirenvSettingsEditor.java
package com.github.direnvloader;

import com.intellij.execution.configurations.RunConfigurationBase;
import com.intellij.openapi.options.SettingsEditor;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

// Run Configuration 설정 패널 UI (체크박스 2개)
public class DirenvSettingsEditor<T extends RunConfigurationBase<?>> extends SettingsEditor<T> {

    private final JCheckBox enableDirenvCheckBox = new JCheckBox("Enable Direnv");
    private final JCheckBox trustEnvrcCheckBox = new JCheckBox("Trust .envrc (auto direnv allow)");
    private final JPanel panel = new JPanel();

    public DirenvSettingsEditor() {
        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
        panel.add(enableDirenvCheckBox);
        panel.add(trustEnvrcCheckBox);
    }

    @Override
    protected void resetEditorFrom(@NotNull T configuration) {
        DirenvSettings settings = configuration.getCopyableUserData(DirenvSettings.KEY);
        if (settings != null) {
            enableDirenvCheckBox.setSelected(settings.isEnabled());
            trustEnvrcCheckBox.setSelected(settings.isTrust());
        } else {
            enableDirenvCheckBox.setSelected(false);
            trustEnvrcCheckBox.setSelected(false);
        }
    }

    @Override
    protected void applyEditorTo(@NotNull T configuration) {
        DirenvSettings settings = new DirenvSettings(
                enableDirenvCheckBox.isSelected(),
                trustEnvrcCheckBox.isSelected()
        );
        configuration.putCopyableUserData(DirenvSettings.KEY, settings);
    }

    @NotNull
    @Override
    protected JComponent createEditor() {
        return panel;
    }
}

File: src/main/kotlin/com/ddoong2/direnvloader/DirenvSettingsEditor.kt
package com.ddoong2.direnvloader

import com.intellij.execution.configurations.RunConfigurationBase
import com.intellij.openapi.options.SettingsEditor
import javax.swing.BoxLayout
import javax.swing.JCheckBox
import javax.swing.JComponent
import javax.swing.JPanel

// Run Configuration 설정 패널 UI (체크박스 2개)
class DirenvSettingsEditor<T : RunConfigurationBase<*>> : SettingsEditor<T>() {

    private val enableDirenvCheckBox = JCheckBox("Enable Direnv")
    private val trustEnvrcCheckBox = JCheckBox("Trust .envrc (auto direnv allow)")
    private val panel = JPanel().apply {
        layout = BoxLayout(this, BoxLayout.Y_AXIS)
        add(enableDirenvCheckBox)
        add(trustEnvrcCheckBox)
    }

    override fun resetEditorFrom(configuration: T) {
        val settings = configuration.getCopyableUserData(DirenvSettings.KEY)
        enableDirenvCheckBox.isSelected = settings?.enabled ?: false
        trustEnvrcCheckBox.isSelected = settings?.trust ?: false
    }

    override fun applyEditorTo(configuration: T) {
        val settings = DirenvSettings(
            enabled = enableDirenvCheckBox.isSelected,
            trust = trustEnvrcCheckBox.isSelected,
        )
        configuration.putCopyableUserData(DirenvSettings.KEY, settings)
    }

    override fun createEditor(): JComponent = panel
}
